qiskit/visualization/gate_map.py

```diff
@@ -338,14 +353,182 @@ def plot_gate_map(
+        [2, 12],
+        [2, 13],
+        [2, 14],
+        [3, 2],
+        [3, 6],
+        [3, 10],
+        [3, 14],
+        [4, 0],
+        [4, 1],
+        [4, 2],
+        [4, 3],
+        [4, 4],
+        [4, 5],
+        [4, 6],
+        [4, 7],
+        [4, 8],
+        [4, 9],
+        [4, 10],
+        [4, 11],
+        [4, 12],
+        [4, 13],
+        [4, 14],
+        [5, 0],
+        [5, 4],
+        [5, 8],
+        [5, 12],
+        [6, 0],
+        [6, 1],
+        [6, 2],
+        [6, 3],
+        [6, 4],
+        [6, 5],
+        [6, 6],
+        [6, 7],
+        [6, 8],
+        [6, 9],
+        [6, 10],
+        [6, 11],
+        [6, 12],
+        [6, 13],
+        [6, 14],
+        [7, 2],
+        [7, 6],
+        [7, 10],
+        [7, 14],
+        [8, 0],
+        [8, 1],
+        [8, 2],
+        [8, 3],
+        [8, 4],
+        [8, 5],
+        [8, 6],
+        [8, 7],
+        [8, 8],
+        [8, 9],
+        [8, 10],
+        [8, 11],
+        [8, 12],
+        [8, 13],
+        [8, 14],
+        [9, 0],
+        [9, 4],
+        [9, 8],
+        [9, 12],
+        [10, 0],
+        [10, 1],
+        [10, 2],
+        [10, 3],
+        [10, 4],
+        [10, 5],
+        [10, 6],
+        [10, 7],
+        [10, 8],
+        [10, 9],
+        [10, 10],
+        [10, 11],
+        [10, 12],
+        [10, 13],
+        [10, 14],
+        [11, 2],
+        [11, 6],
+        [11, 10],
+        [11, 14],
+        [12, 1],
+        [12, 2],
+        [12, 3],
+        [12, 4],
+        [12, 5],
+        [12, 6],
+        [12, 7],
+        [12, 8],
+        [12, 9],
+        [12, 10],
+        [12, 11],
+        [12, 12],
+        [12, 13],
+        [12, 14],
+    ]
+
+    backend_version = _get_backend_interface_version(backend)
+    if backend_version <= 1:
+        from qiskit.transpiler.coupling import CouplingMap
+
+        if backend.configuration().simulator:
+            raise QiskitError("Requires a device backend, not simulator.")
+        config = backend.configuration()
+        num_qubits = config.n_qubits
+        coupling_map = CouplingMap(config.coupling_map)
+        name = backend.name()
+    else:
+        num_qubits = backend.num_qubits
+        coupling_map = backend.coupling_map
+        name = backend.name
+    if qubit_coordinates is None and ("ibm" in name or "fake" in name):
+        qubit_coordinates = qubit_coordinates_map.get(num_qubits, None)
+
+    if qubit_coordinates is None:
+        # Replace with planar_layout() when retworkx offers it
+        qubit_coordinates_rx = rx.spring_layout(coupling_map.graph, seed=1234)
+        scaling_factor = 10 ** int(math.log10(num_qubits) + 1)
+        qubit_coordinates = [
+            (
+                int(scaling_factor * qubit_coordinates_rx[i][0]),
+                int(scaling_factor * qubit_coordinates_rx[i][1]),
+            )
+            for i in range(num_qubits)
+        ]
+
+    if any(x[0] < 1 or x[1] < 1 for x in qubit_coordinates):
```

Comment:
Should these check if < 0 instead? If the smallest value is 0, won't `qubit_coordinates` be unchanged?

Reply:
Oh, yep that's a good catch. Not sure why I put 1 here, it definitely is supposed to be 0 as this is trying to find negative values and shift everything up so it's all >=0.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7814/commits/01fb3a03e30f59338c032f085ceb8314ed7a4ea8